backend/tests/v5/e2e/routes/users.test.js

```diff
@@ -152,16 +156,225 @@ const testGetUsername = () => {
+			const res = await testSession.put('/v5/user/').send(data).expect(templates.invalidArguments.status);
+			expect(res.body.code).toEqual(templates.invalidArguments.code);
+		});
+
+		test('should update the profile if the user is logged in', async () => {
+			const data = { firstName: 'newName' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			await testSession.put('/v5/user/').send(data).expect(200);
+		});
+
+		test('should update the profile and change password if the user is logged in', async () => {
+			const data = { firstName: 'newName', oldPassword: testUser.password, newPassword: 'Passport123!' };
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			await testSession.put('/v5/user/').send(data).expect(200);
+			await testSession.put('/v5/user/').send({ oldPassword: 'Passport123!', newPassword: testUser.password });
+		});
+	});
+};
+
+const testGenerateApiKey = () => {
+	describe('Generate and assign Api key to the logged in user', () => {
+		test('should fail if the user is not logged in', async () => {
+			const res = await agent.post('/v5/user/key').expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the user has a session via an API key', async () => {
+			const res = await agent.post(`/v5/user/key?key=${testUser.apiKey}`)
+				.expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should create and return a new Api key if the user is logged in', async () => {
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			const res = await testSession.post('/v5/user/key').expect(200);
+			const userProfileRes = await testSession.get('/v5/user');
+			expect(res.body).toEqual({ apiKey: userProfileRes.body.apiKey });
+		});
+	});
+};
+
+const testDeleteApiKey = () => {
+	describe('Delete the Api key of the logged in user', () => {
+		test('should fail if the user is not logged in', async () => {
+			const res = await agent.delete('/v5/user/key').expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should fail if the user has a session via an API key', async () => {
+			const res = await agent.delete(`/v5/user?key=${testUser.apiKey}`)
+				.expect(templates.notLoggedIn.status);
+			expect(res.body.code).toEqual(templates.notLoggedIn.code);
+		});
+
+		test('should delete the Api key if the user is logged in', async () => {
+			await testSession.post('/v5/login/').send({ user: testUser.user, password: testUser.password });
+			await testSession.delete('/v5/user/key').expect(200);
+			const userProfileRes = await testSession.get('/v5/user');
+			expect(userProfileRes.body.apiKey).toEqual(undefined);
```

Comment:
can we also do a check that the api key will not longer be valid? (i.e. do a get request with it and see)

Reply:
you mean testuser.ApiKey or userProfileRes.body.apiKey ? cause the second is undefined

Reply:
i mean once you call delete on the key, use that key to do a http request again and see if you get an error. So it would be the api key that you would get from the endpoint prior to you deleting it